Remove progress package (spinner)

gauron99 opened the pull request:
# Changes
- :wastebasket: Remove feature or internal logic - remove spinner progress package, convert to printing to stderr instead as a first step to changing to a new logging package.

/kind removal
Fixes #1893 
Fixes #1513

gauron99 commented:
cc @lkingland

matejvasek commented:
@gauron99 please try running `TestOnClusterBuild/pack` locally.

matejvasek commented on pkg/pipelines/tekton/pipelines_integration_test.go:
```diff
@@ -114,35 +113,6 @@ func (p pl) log(args ...any) {
-			return
-		}
-		u := strings.TrimSpace(parts[1])
-		p.urlChan <- u
```
Removal of this causes timeout of test since it never gets message that function has been deployed. The are additional changes required to make test work again.

matejvasek replied:
For instance we could add `io.Reader`'s for error/normal outputs into the `PipelinesProvider` structure and print to them instead to `os.Std[out|err]`. The test then could pass pipe when constructing `PipelinesProvider` and inspect output for occurrence of `exposed at URL:\n(\w)`.

wdyt @lkingland

matejvasek replied:
@gauron99 other maybe cleaner way would be to also return `fn.Instance` from the `Run()` method.

matejvasek replied:
The point is that `Run()` should somehow return url of deployed function. Be it via return value or output.

matejvasek replied:
So you can try invoking it.

lkingland reviewed (commented):
Looks good so far!

gauron99 commented:
@matejvasek @lkingland PTAL guys, removed the goroutines and added the return value for URL as string. Not sure how to handle the structs in other files, where Run method is used so please let me know on that. I ran the test individually and collectively with `make test` and both seem to work fine

matejvasek commented on pkg/pipelines/tekton/pipelines_integration_test.go:
```diff
@@ -57,18 +55,22 @@ func TestOnClusterBuild(t *testing.T) {
 			f := createSimpleGoProject(t, ns)
 			f.Build.Builder = test.Builder
 
 			go func() {
```
Give we do not retrieve asynchronously we could simplify this code and get rid of channels and goroutines.

matejvasek replied:
Otherwise looks good.

matejvasek commented:
/approve
/lgtm
